models/aeif_cbvg_2010.cpp

```diff
@@ -0,0 +1,657 @@
+  const double& u_bar_bar = y[ S::U_BAR_BAR ];
+
+  const double I_spike = ( node.P_.g_L * node.P_.Delta_T
+    * std::exp( ( V - V_T ) / node.P_.Delta_T ) );
+
+  // dv/dt
+  f[ S::V_M ] = ( is_refractory || is_clamped )
+    ? 0.0
+    : ( -node.P_.g_L * ( V - node.P_.E_L ) + I_spike - w + z + node.P_.I_e
+        + node.B_.I_stim_ ) / node.P_.C_m;
+
+  // Adaptation current w.
+  f[ S::W ] =
+    is_clamped ? 0.0 : ( node.P_.a * ( V - node.P_.E_L ) - w ) / node.P_.tau_w;
+
+  f[ S::Z ] = -z / node.P_.tau_z;
+
+  f[ S::V_T ] = -( V_T - node.P_.V_T_rest ) / node.P_.tau_V_T;
+
+  f[ S::U_BAR_PLUS ] = ( -u_bar_plus + V ) / node.P_.tau_plus;
+
+  f[ S::U_BAR_MINUS ] = ( -u_bar_minus + V ) / node.P_.tau_minus;
+
+  f[ S::U_BAR_BAR ] = ( -u_bar_bar + u_bar_minus ) / node.P_.tau_bar_bar;
+
+  return GSL_SUCCESS;
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::aeif_cbvg_2010::Parameters_::Parameters_()
+  : V_peak_( 0.0 )       // mV
+  , V_reset_( -60.0 )    // mV
+  , t_ref_( 0.0 )        // ms
+  , g_L( 30.0 )          // nS
+  , C_m( 281.0 )         // pF
+  , E_L( -70.6 )         // mV
+  , Delta_T( 2.0 )       // mV
+  , tau_w( 144.0 )       // ms
+  , tau_z( 40.0 )        // ms
+  , tau_V_T( 50.0 )      // ms
+  , V_T_max( 30.4 )      // mV
+  , V_T_rest( -50.4 )    // mV
+  , tau_plus( 7.0 )      // ms
+  , tau_minus( 10.0 )    // ms
+  , tau_bar_bar( 500.0 ) // ms
+  , a( 4.0 )             // nS
+  , b( 80.5 )            // pA
+  , I_sp( 400.0 )        // pA
+  , I_e( 0.0 )           // pA
+  , gsl_error_tol( 1e-6 )
+  , t_clamp_( 2.0 )  // ms
+  , V_clamp_( 33.0 ) // mV
+{
+}
+
+nest::aeif_cbvg_2010::State_::State_( const Parameters_& p )
+  : r_( 0 )
+  , clamp_r_( 0 )
+{
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+  {
+    y_[ i ] = 0;
+  }
+  y_[ V_M ] = p.E_L;
+  y_[ V_T ] = p.V_T_rest;
+  y_[ U_BAR_PLUS ] = p.E_L;
+  y_[ U_BAR_MINUS ] = p.E_L;
+  y_[ U_BAR_BAR ] = p.E_L;
+}
+
+nest::aeif_cbvg_2010::State_::State_( const State_& s )
+  : r_( s.r_ )
+  , clamp_r_( s.clamp_r_ )
+{
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+  {
+    y_[ i ] = s.y_[ i ];
+  }
+}
+
+nest::aeif_cbvg_2010::State_& nest::aeif_cbvg_2010::State_::operator=(
+  const State_& s )
+{
+  assert( this != &s ); // would be bad logical error in program
+
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+  {
+    y_[ i ] = s.y_[ i ];
+  }
+  r_ = s.r_;
+  clamp_r_ = s.clamp_r_;
+  return *this;
+}
+
+/* ----------------------------------------------------------------
+ * Paramater and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::aeif_cbvg_2010::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::C_m, C_m );
+  def< double >( d, names::V_T_max, V_T_max );
```

Comment:
I am wondering about this... I guess ``V_T`` is used for the clopath model, but for the AdExp model, it is usually named ``V_th``...

Reply:
This seems to cause some confusion. We agree that `V_th` seems to be the better name and changed the name with 511fe48 (this commit will be merged soon).